sympy/vector/scalar.py

```diff
@@ -13,11 +13,17 @@ class BaseScalar(Symbol):
     """
 
     def __new__(cls, name, index, system, pretty_str, latex_str):
-        name = str(name)
-        pretty_str = str(pretty_str)
-        latex_str = str(latex_str)
         from sympy.vector.coordsysrect import CoordSysCartesian
-        obj = super(BaseScalar, cls).__new__(cls, name)
+        if isinstance(name, Symbol):
+            name = name.name
+        if isinstance(pretty_str, Symbol):
+            pretty_str = pretty_str.name
+        if isinstance(latex_str, Symbol):
+            latex_str = latex_str.name
+
+        index = _sympify(index)
+        system = _sympify(system)
+        obj = super(BaseScalar, cls).__new__(cls, Symbol(name), index, system, Symbol(pretty_str), Symbol(latex_str))
```

Comment:
https://github.com/asmeurer/sympy/blob/6ded1d78dd0bfb457cf2036416c35bce9daad8d1/sympy/vector/coordsysrect.py#L17

`variable_names` is passed as argument to `CoordSysCartesian`, but not registered as _args_ there. It gets passed as `name` here. I think it should be stored in `CoordSysCartesian`'s _args_, so that `name` would be available from `system` here.

That's the way @Krastanov did it in _sympy.diffgeom_, it looks neater.

Same argument for vector fields.